share/shwap/p2p/shrex/shrex_getter/shrex.go

```diff
@@ -143,105 +95,164 @@ func (sg *Getter) Stop(ctx context.Context) error {
 	return sg.archivalPeerManager.Stop(ctx)
 }
 
-func (sg *Getter) GetSamples(context.Context, *header.ExtendedHeader, []shwap.SampleCoords) ([]shwap.Sample, error) {
-	return nil, fmt.Errorf("getter/shrex: GetShare %w", shwap.ErrOperationNotSupported)
-}
+func (sg *Getter) GetSamples(
+	ctx context.Context,
+	header *header.ExtendedHeader,
+	coords []shwap.SampleCoords,
+) ([]shwap.Sample, error) {
+	// short circuit if the data root is empty
+	if header.DAH.Equals(share.EmptyEDSRoots()) {
+		return []shwap.Sample{}, nil
+	}
 
-func (sg *Getter) GetRow(_ context.Context, _ *header.ExtendedHeader, _ int) (shwap.Row, error) {
-	return shwap.Row{}, fmt.Errorf("getter/shrex: GetRow %w", shwap.ErrOperationNotSupported)
+	var err error
+	ctx, span := tracer.Start(ctx, "shrex/get-samples")
+	defer func() {
+		utils.SetStatusAndEnd(span, err)
+	}()
+
+	requests := make([]shwap.SampleID, len(coords))
+	for i, coord := range coords {
+		sID, err := shwap.NewSampleID(header.Height(), coord, len(header.DAH.RowRoots))
+		if err != nil {
+			return nil, fmt.Errorf("failed to create a sampleID from the coordinates: %w", err)
+		}
+		requests[i] = sID
+	}
+
+	samples := make([]shwap.Sample, len(requests))
+	errGroup, ctx := errgroup.WithContext(ctx)
+	errGroup.SetLimit(len(requests))
+
+	for i, request := range requests {
+		logger := log.With(
+			"source", "shrex_getter",
+			"request_type", request.Name(),
+			"hash", header.DAH.String(),
+			"rowIndex", request.RowIndex,
+			"colIndex", request.ShareIndex,
+		)
+		errGroup.Go(func() error {
+			req := func(ctx context.Context, peer libpeer.ID) (int64, error) {
+				return sg.client.Get(ctx, &request, &samples[i], peer)
+			}
+			verify := func() error {
+				if samples[i].IsEmpty() {
+					return errors.New("nil response")
+				}
+				return samples[i].Verify(header.DAH, request.RowIndex, request.ShareIndex)
+			}
+			return sg.executeRequest(ctx, logger, header, request.Name(), req, verify)
+		})
+	}
+	if err = errGroup.Wait(); err != nil {
+		return nil, err
+	}
+	return samples, nil
```

Comment:
Is it intended that this will not return partial response for sampling? We used to returned successful samples so next retry attempt don't have to re-request them